docker-compose.yml

```
@@ -0,0 +1,27 @@
+version: '3.0'
+
+services:
+  tests:
+    build: .
+    container_name: php-imap
+    volumes:
+      - .:/project
+    command: bash -c "sleep 1s && /project/vendor/bin/phpunit /project/tests"
```

```suggestion
    command: bash -c "sleep 1s && cd /project && vendor/bin/phpunit"
```
`CD`ing into the directory gives us the benefit of reading the committed `phpunit.xml.dist`

Is the `sleep 1s` really necessary?

---

the sleep is necessary because the imap server doesn't respond immediately.

Using php 7.0 I get the following error:

`This version of PHPUnit is supported on PHP 7.1 and PHP 7.2.
You are using PHP 7.0.32 (/usr/local/bin/php).`

Would it be a problem to just lower the phpunit version?

---

The problem is not PHPUnit, but the fact that `composer install` was run into an environment different from docker, and thus a different PHPUnit version was chosen.

The correct solution is copying current directory into the docker image, and then running `composer install` at every run, to ensure a proper image initialization without touching the host non-committed files.

Could you give it a try?

---

> If every time I want to run the tests I have to rebuild the docker image and run composer install, the whole thing just becomes slow and clumsy.

The docker image wouldn't need a rebuild.

I've done some tests: if a developer uses Docker to test this library, it is very unlikely that tests are run outside Docker.

So I tried to add composer to the Dockerfile, and run composer install at every `docker-compose run tests` without protecting the hosts for changes. The dependancies are updated the first run, and then the other runs are quick since no update is needed, `vendor/` stay aligned with Docker environments requirements.

To me this is practical enough. These are the files I've used:

#### Dockerfile
```dockerfile
FROM php:7.0

RUN runtimeDeps=" \
        curl \
        git \
        zip \
        libc-client-dev \
        libkrb5-dev \
        libxml2-dev \
    " \
    && apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y $runtimeDeps \
    && docker-php-ext-configure imap --with-kerberos --with-imap-ssl \
    && docker-php-ext-install iconv mbstring imap \
    && rm -r /var/lib/apt/lists/*

RUN curl -sS https://getcomposer.org/installer | php -- --install-dir=/usr/local/bin --filename=composer
```

#### docker-compose.yml

```yaml
version: '3.0'

services:
  tests:
    build: .
    container_name: php-imap
    volumes:
      - .:/project
    command: bash -c "cd /project && rm composer.lock && composer install && vendor/bin/phpunit"
    environment:
      - IMAP_SERVER_NAME=imap.server
      - IMAP_SERVER_PORT=993
      - IMAP_USERNAME=test@test.test
      - IMAP_PASSWORD=p4ssword
    depends_on:
      - imap
  imap:
    image: antespi/docker-imap-devel:latest
    container_name: imap-server
    environment:
    - MAILNAME=test.test
    - MAIL_ADDRESS=test@test.test
    - MAIL_PASS=p4ssword
    networks:
      default:
        aliases:
        - 'imap.server'
```